libs/components/core/src/lib/modules/resize-observer/resize-observer-media-query.service.ts

```diff
@@ -0,0 +1,97 @@
+import { ElementRef, Injectable, OnDestroy } from '@angular/core';
+
+import { Observable, ReplaySubject, Subject, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { SkyMediaBreakpoints } from '../media-query/media-breakpoints';
+import { SkyMediaQueryListener } from '../media-query/media-query-listener';
+
+import { SkyResizeObserverService } from './resize-observer.service';
+
+/**
+ * Acts like `SkyMediaQueryService` for a container element, emitting the same responsive breakpoints.
+ */
+@Injectable()
+export class SkyResizeObserverMediaQueryService implements OnDestroy {
+  /**
+   * Returns the current breakpoint.
+   */
+  public get current(): SkyMediaBreakpoints {
+    return this._current;
+  }
+
+  private currentSubject = new ReplaySubject<SkyMediaBreakpoints>(1);
+
+  private _current: SkyMediaBreakpoints;
+  private _observable: Observable<ResizeObserverEntry>;
+  private _breakpoints: {
+    check: (width: number) => boolean;
+    name: SkyMediaBreakpoints;
+  }[] = [
+    {
+      check: (width: number) => width <= 767,
+      name: SkyMediaBreakpoints.xs,
+    },
+    {
+      check: (width: number) => width > 767 && width <= 991,
+      name: SkyMediaBreakpoints.sm,
+    },
+    {
+      check: (width: number) => width > 991 && width <= 1199,
+      name: SkyMediaBreakpoints.md,
+    },
+    {
+      check: (width: number) => width > 1199,
+      name: SkyMediaBreakpoints.lg,
+    },
+  ];
+  private stopListening = new Subject<void>();
+
+  constructor(private resizeObserverService: SkyResizeObserverService) {}
+
+  public ngOnDestroy(): void {
+    this.removeListeners();
+    this.stopListening.complete();
+    this.currentSubject.complete();
+  }
+
+  /**
+   * Sets the container element to watch. The `SkyResizeObserverMediaQueryService` will only observe one element at a
+   * time. Any previous subscriptions will be unsubscribed when a new element is observed.
+   * @param element
+   */
+  public observe(element: ElementRef): void {
+    this.stopListening.next();
```

Comment:
Should we reset `current` whenever we observe a new element? We call `checkBreakPointChange` but that won't emit if a second element is observed that had the same breakpoint as a previously observed element. That is maybe ok? I just wanted to clarify.